// karma-polymer-test/gruntfile.js
module.exports = function(grunt) {
  'use strict';

  // Project Configuration
  grunt.initConfig({
    // Metadata
    pkg: grunt.file.readJSON('package.json'),
    bower: grunt.file.exists('.bowerrc') && grunt.file.readJSON('.bowerrc') || {directory:'bower_components'},

    // Task configuration
    'bower-install-simple': {
        options: {
            forceLatest: true,
            color: true,
            production: false,
            directory: '<%= bower.directory %>'
        }
    },
    yuidoc: {
      compile: {
        name: '<%= pkg.title %>',
        description: '<%= pkg.description %>',
        version: '<%= pkg.version %>',
        url: '<%= pkg.homepage %>',
        options: {
          exclude: 'build,<%= bower.directory %>',
          extension: '.js,.html',
          tabtospace: 2,
          paths: '.',
          outdir: 'build/doc',
          themedir: 'node_modules/yuidoc-bootstrap-theme',
          helpers: ['node_modules/yuidoc-bootstrap-theme/helpers/helpers.js']
        }
      }
    },
    jshint: {
      options: {
        ignores: 'build,<%= bower.directory %>',
        extract: 'auto',
        browser: true,
        node: true,
        devel: true,
        camelcase: true,
        curly: true,
        forin: true,
        immed: true,
        eqeqeq: true,
        quotmark: 'single',
        strict: 'true',
        nonew: true,
        noarg: true,
        nonbsp: 'true',
        noempty: 'true',
        undef: 'true',
        unused: 'true',
        trailing: 'true',
        indent: 2,
        maxlen: 120,
        maxparams: 4,
        maxdepth: 3,
        maxcomplexity: 5,
        globals: {
          'Polymer': true
        }
      },
      all: ['*.html', '*.js']
    }
  });

  // Plugins
  grunt.loadNpmTasks('grunt-bower-install-simple');
  grunt.loadNpmTasks('grunt-contrib-yuidoc');
  grunt.loadNpmTasks('grunt-contrib-jshint');
  grunt.loadNpmTasks('grunt-release');

  // Some aliases
  grunt.registerTask('bower-install', [ 'bower-install-simple' ]);

  // Task registration
  grunt.registerTask('default', ['build']);
  grunt.registerTask('docs', ['bower-install', 'yuidoc']);
  grunt.registerTask('sca', ['bower-install', 'jshint']);
  grunt.registerTask('build', ['sca']);
};
